xoa/internal/hashcode.go

```diff
@@ -29,8 +29,12 @@ func String(s string) int {
 func Strings(strings []string) string {
 	var buf bytes.Buffer
 
-	for _, s := range strings {
-		buf.WriteString(fmt.Sprintf("%s-", s))
+	for i, s := range strings {
+		var format = "%s"
+		if i != len(strings)-1 {
+			format = fmt.Sprint(format, "-")
+		}
+		buf.WriteString(fmt.Sprintf(format, s))
```

Comment:
Is there a reason the previous implementation didn't work? I copy and pasted this directly from v1 of the terraform-plugin-sdk package so while this functionally doesn't make a difference, I would prefer to keep it identical or migrate to something supported (given this is deprecated code).

Reply:
```

Reply:
Apply this patch (file below) to reset the changes to CRC hash calculation 
These changes were already committed in #161 with the revisions 267cd30 and 8a31b0c

```
Index: xoa/data_source_vms.go
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/xoa/data_source_vms.go b/xoa/data_source_vms.go
--- a/xoa/data_source_vms.go	(revision 470c245132d51d70fee07e45aa0ba9a6a2c37b73)
+++ b/xoa/data_source_vms.go	(revision 8a31b0c725d48cbe88cf930fd0c695285e7ac72d)
@@ -1,11 +1,13 @@
 package xoa
 
 import (
+	"fmt"
+	"github.com/ddelnano/terraform-provider-xenorchestra/xoa/internal"
 	"log"
+	"strconv"
 	"strings"
 
 	"github.com/ddelnano/terraform-provider-xenorchestra/client"
-	"github.com/ddelnano/terraform-provider-xenorchestra/xoa/internal"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
@@ -51,7 +53,10 @@
 	if err = d.Set("vms", vmToMapList(vms)); err != nil {
 		return err
 	}
-	d.SetId(internal.Strings([]string{searchVm.PowerState, searchVm.PoolId, searchVm.Host}))
+	hash := internal.String(fmt.Sprintf(
+		"%s-%s-%s-", searchVm.PowerState, searchVm.PoolId, searchVm.Host,
+	))
+	d.SetId(strconv.Itoa(hash))
 	return nil
 
 }
Index: xoa/internal/hashcode.go
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/xoa/internal/hashcode.go b/xoa/internal/hashcode.go
--- a/xoa/internal/hashcode.go	(revision 470c245132d51d70fee07e45aa0ba9a6a2c37b73)
+++ b/xoa/internal/hashcode.go	(revision 267cd3075b2c6a42607e1fce01202b15162b04d4)
@@ -29,12 +29,8 @@
 func Strings(strings []string) string {
 	var buf bytes.Buffer
 
-	for i, s := range strings {
-		var format = "%s"
-		if i != len(strings)-1 {
-			format = fmt.Sprint(format, "-")
-		}
-		buf.WriteString(fmt.Sprintf(format, s))
+	for _, s := range strings {
+		buf.WriteString(fmt.Sprintf("%s-", s))
 	}
 
 	return fmt.Sprintf("%d", String(buf.String()))

```